### PR TITLE
gui: render instance text centered, easier to read

### PR DESCRIPTION
**Pull request by oharboe**
Before:

image

image

After:

image

image

### CODE REVIEW

**Comment by gadfort** [comment]
@oharboe can you provide before and after images of this change?

**Comment by gadfort** [comment]
How does this look on standard cells?

**Comment by gadfort** [comment]
@oharboe I think for the macros it helps, but the standard cells I don't know if that is an improvement, it's actually harder to determine the boundaries of the cells (since it used to be where the text starts).

**Comment by oharboe** [comment]
> @oharboe I think for the macros it helps, but the standard cells I don't know if that is an improvement, it's actually harder to determine the boundaries of the cells (since it used to be where the text starts).

Is there a way, from within that fn to determine when this should be done and not done?

**Comment by oharboe** [comment]
> You should be able to check the type of the cell (`inst->getMaster()->isMacro()`) or something like that. I suspect pads and cover cells also benefit from something like this.

Fixed.

**Comment by maliberty** [comment]
In the after picture, one of the srams has its label horizontally while the other three are vertical.  What explains that?

File: src/gui/src/renderThread.cpp [hunk 1]
```diff
@@ -697,13 +700,23 @@ bool RenderThread::drawTextInBBox(const QColor& text_color,
   painter->scale(scale_adjust, -scale_adjust);
   if (do_rotate) {
     text_bounding_box = font_metrics.boundingRect(name);
+    auto xOffset = (bbox_in_px.height() - text_bounding_box.width()) / 2;
+    auto yOffset = (bbox_in_px.width() - text_bounding_box.height()) / 2;
```
Comment (maliberty) [inline src/gui/src/renderThread.cpp:704]: Move inside 'if (center)' since they are only needed at that scope.  Also const.  Same in the else block.

**Comment by oharboe** [comment]
@maliberty horizontal vs. vertical text direction is chosen based on the number of pixels wide the text is. the macros have different labels, so if you zoom just right, you can illustrate horizontal and vertical text in the same screenshot

**Pull request merged**